rudder-lang/src/ast/context.rs

```diff
@@ -82,20 +82,62 @@ impl<'src> VarContext<'src> {
         enum1: Token<'src>,
         value: Option<Token<'src>>,
     ) -> Result<()> {
-        self.new_var(upper_context, name)?;
+        self.check_new_var(upper_context, name)?;
         self.variables.insert(name, VarKind::Enum(enum1, value));
         Ok(())
     }
 
-    /// Create a generic variable in tis context.
+    fn push_or_err_rec(name: Token<'src>, stored_ctx: &mut Value<'src>, new_var: &Value<'src>) -> Result<()> {
```

Comment:
This function name is a bit meaningless, can you comment it and rename it ?

Reply:
improved it

Reply:
improved it